kernel_gateway/services/kernels/handlers.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+import notebook.services.kernels.handlers as notebook_handlers
+
+class TokenAuthorizationMixin(object):
+    '''
+    Adds token authentication to tornado.web.RequestHandlers.
+    '''
+    def prepare(self):
+        '''
+        If kg_auth_token is present in the application settings,
+        raises 401 if the request Authorization header does not contain
+        the value "Token <kg_auth_token>".
+        '''
+        server_token = self.settings.get('kg_auth_token')
+        if server_token:
+            client_token = self.request.headers.get('Authorization')
+            if client_token != 'Token %s' % server_token:
```

Comment:
I'm not sure what the standard practice, but from working with the GitHub API I've gotten used to `Token` being lowercase `token`:

```
curl -H "Authorization: token 6c0b8049fa7de55e31becc84dc970498" https://api.github.com
```

(that's a randomly generated token by the way)